### PR TITLE
Gradle 4.8-rc-1 compat fix - use def when finding the buildType

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal
Gradle 4.8-rc-1 changes `project.android.buildTypes.store` to return an `Iterable` rather than a `Set`, which can lead to a runtime error within the plugin. This changeset prevents the runtime error by using `def` instead.

### Changed
Altered affected types from specific `Set` implementation to `def`.

## Tests

Mazerunner was run locally after running `./gradlew wrapper --gradle-version 4.8-rc-1` in each of the fixture projects.

### Alternative Approaches

We could use `Iterable` instead of `def`.

### Linked issues

#109 

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [x] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [x] Initial review of the intended approach, not yet feature complete
  - [x] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [x] Consistency across platforms for structures or concepts added or modified
- [x] Consistency between the changeset and the goal stated above
- [x] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [x] Usage friction - is the proposed change in usage cumbersome or complicated?
- [x] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [x] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [x] Thoroughness of added tests and any missing edge cases
- [x] Idiomatic use of the language

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+## 3.2.7 (TBD)
+
+### Bug fixes
+
+* Gradle 4.8-rc-1 compat fix - use def when finding the buildType
```
Comment (kattrali) [inline CHANGELOG.md]: This summary seems unclear. It should say is happening from a user perspective, e.g. "Add compatibility with Gradle 4.8-rc1"

**Comment by fractalwrench** [comment]
Looks like whatever is wrong with CI is wrong on master also.
